src/parcsr_mv/par_csr_matvec.c

```diff
@@ -146,8 +146,11 @@ hypre_ParCSRMatrixMatvecOutOfPlace( HYPRE_Complex       alpha,
    {
       if (!hypre_ParCSRCommPkgTmpData(comm_pkg))
       {
-         /* hypre_ParCSRCommPkgTmpData(comm_pkg) = hypre_TAlloc(HYPRE_Complex, num_cols_offd, HYPRE_MEMORY_DEVICE); */
+#if 1
+         hypre_ParCSRCommPkgTmpData(comm_pkg) = hypre_TAlloc(HYPRE_Complex, num_cols_offd, HYPRE_MEMORY_DEVICE);
+#else
          hypre_ParCSRCommPkgTmpData(comm_pkg) = _hypre_TAlloc(HYPRE_Complex, num_cols_offd, hypre_MEMORY_DEVICE);
+#endif
```

Comment:
What does _hypre_TAlloc do compared to hypre_TAlloc?

Reply:
`hypre_TAlloc` is the macro for
```
void * hypre_MAlloc(size_t size, HYPRE_MemoryLocation location);
```
Similarly, `_hypre_TAlloc` is for
```
void * _hypre_MAlloc(size_t size, hypre_MemoryLocation location);
```
The former one is for "conceptual" memory spaces (HYPRE_MemoryLocation) and the latter one is for physical spaces (hypre_MemoryLocation). Both are supposed for developers. In the case of hypre with UVM, the latter one can be used for allocating device memory since the former one will return UVM. Since we have removed dependencies of UVM, I changed them back to `hypre_TAlloc`. The name starting with `-` suggests that it is really an internal function. One needs to know what they are doing with it.